cypress/integration/chi-js/dropdown.spec.js

```diff
@@ -0,0 +1,213 @@
+            .find('button#dropdown-1.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.visible')
+            .should('have.class', '-active')
+        });
+    
+        it('Dropdown menu should close when dropdown trigger is clicked twice', () => {  
+            cy.get('.chi-dropdown')
+            .find('button#dropdown-1.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.not.visible')
+        });
+    })
+
+    describe('Open and closing functionality test for dropdown target', function() {
+        it('Dropdown menu should open when dropdown trigger is clicked', () => {  
+            cy.get('button#dropdown-2.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .should('be.visible')
+            .should('have.class', '-active')
+            
+        });
+
+        it('Dropdown menu should close when dropdown trigger is clicked', () => {  
+            cy.get('button#dropdown-2.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+        });
+    })
+
+    describe('Functionality test for dropdown on hover', function () {
+          it('Dropdown menu should open when hovered over ', () => {
+            cy.get('.chi-dropdown.chi-dropdown__hover')
+            .find('button.chi-dropdown__trigger')
+            .find('+.chi-dropdown__menu')
+            .invoke('show')
+            .wait(550)
+            .should('be.visible')
+            // .should('have.class', '-active')
+          })
+
+          it('Dropdown menu should close when dropdown trigger is clicked', () => {  
+            cy.get('.chi-dropdown.chi-dropdown__hover')
+            .find('button.chi-dropdown__trigger')
+            .find('+.chi-dropdown__menu')
+            .invoke('hide')
+            .wait(550)
+            .should('be.not.visible')
+          });
+    })
+
+    describe('Open and closing functionality test for animated dropdown', function() {
+        it('Dropdown menu should open when dropdown trigger is clicked', () => {  
+            cy.get('.chi-dropdown')
+            .find('button#dropdown-animate.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.visible')
+            .should('have.class', '-active')
+        });
+    
+        it('Dropdown menu should close when dropdown trigger is clicked twice', () => {  
+            cy.get('.chi-dropdown')
+            .find('button#dropdown-animate.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.not.visible')
+        });
+    })
+
+    describe('Open and closing functionality test for positioned dropdown', function() {
```

Comment:
Please check out the implementation of a similar test of Chi Popover done by Shintu using an array of available positions of Popover (in this case it would be Dropdown) and checking menu visibility of each one.

Reply:
Changes implemented in latest commit